create-envfile.py

```diff
@@ -121,35 +121,21 @@ def _get_vals_to_replace(args):
                 _jsfile = json.load(_json_file)
 
         _vals_to_replace = {
-            key: val for key, val in vars(args).items() if key not in _config
+            key: _jsfile.get(key, val) for key, val in vars(args).items() if key not in _config
         }
+        tcp = 'https' if eval(_jsfile.get("https", f"args.https")) else 'http'
 
-        _vals_to_replace["http_host"] = "" if _jsfile.get("https", None) or args.https else args.hostname
-        _vals_to_replace["https_host"] = "" if not _jsfile.get("https", None) or args.https else args.hostname
+        _vals_to_replace["http_host"] = _jsfile.get("hostname", args.hostname) if tcp == 'http' else ""
+        _vals_to_replace["https_host"] = _jsfile.get("hostname", args.hostname) if tcp == 'https' else ""
 
-        _vals_to_replace["geoserver_ui"] = (
-            f"http://{args.hostname}" if not _vals_to_replace.get("https_host") else f"https://{args.hostname}"
-        )
-
-        _vals_to_replace["secret_key"] = args.secret_key or "".join(random.choice(_strong_chars) for _ in range(50))
-        _vals_to_replace["letsencrypt_mode"] = (
-            "disabled" if not _vals_to_replace.get("https_host") else "production"
-        )
-        _vals_to_replace["debug"] = False if (_jsfile.get("env_type", "prod") or args.env_type )in ["prod", "test"] else True
-        _vals_to_replace["email"] = args.email or _jsfile.get("email", "")
+        _vals_to_replace["geoserver_ui"] = f"{tcp}://{_jsfile.get('hostname', args.hostname)}"
+        _vals_to_replace["secret_key"] = _jsfile.get("secret_key",args.secret_key) or "".join(random.choice(_strong_chars) for _ in range(50))
+        _vals_to_replace["letsencrypt_mode"] = "disabled" if not _vals_to_replace.get("https_host") else "production"
+        _vals_to_replace["debug"] = False if _jsfile.get("env_type", args.env_type) in ["prod", "test"] else True
+        _vals_to_replace["email"] = _jsfile.get("email", args.email)
 
-        if _vals_to_replace.get("https_host") and not _vals_to_replace["email"]:
+        if tcp == 'https' and not _vals_to_replace["email"]:
             raise Exception("With HTTPS enabled, the email parameter is required")
-        
-        _updated_vals = _vals_to_replace.copy()
```

Comment:
I would rather consider keeping the in-line command with a higher priority instead of the JSON one.
For example, if I just want to change a value for a single creation, is easier to just add the params than changing the JSON.
@giohappy what do you think?

Reply:
This would mean that for values with defaults, JSON values will not be considered. eg `hostname`

Reply:
Also currently I don't think we provide the option of changing one arg since we always use .env.sample which doesn't have previous values for other args.